src/tcp.ts

```diff
@@ -17,17 +30,34 @@ export function createTcpProxy(config: Config, logger: Logger) {
     const bindClientToServer = (client: TcpTarget, server: TcpTarget) => {
         server.socket.once('end', err => {
             logger.log('SOCKET_UNBOUND', `${client.address}:${client.port}`, `${server.address}:${server.port}`, err?.message)
+            // If the connection between the 'client' and 'server' ends, the connection to the 'client' doesn't need to end.
+            // When the 'client' sends its next message, the proxy will create a new connection to the 'server' and proxy the message.
+            // If the 'client' is awaiting a response from the 'server', it will eventually time out and reconnect, and then resend its last message.
         }).once('error', err => {
             logger.log('SOCKET_BIND_ERROR', `${client.address}:${client.port}`, `${server.address}:${server.port}`, err?.message)
             client.socket.end()
+        }).once('timeout', () => {
+            logger.log('SERVER_SOCKET_TIMEOUT', `${client.address}:${client.port}`, `${server.address}:${server.port}`)
+            server.socket.end() // Only end the server socket
         }).once('connect', () => {
             logger.log('SOCKET_BOUND', `${client.address}:${client.port}`, `${server.address}:${server.port}`)
 
+            // Configure keep-alive on 'client' side connection
+            configureSocketKeepAlive(client.socket, config.clientKeepAliveInterval, 'client')
+            // Configure keep-alive on 'server' side connection
+            configureSocketKeepAlive(server.socket, config.serverKeepAliveInterval, 'server')
+
             server.socket.pipe(client.socket)
-            bindTargetToLogger(server, client, logger);
+            bindTargetToLogger(server, client, logger)
 
             client.socket.pipe(server.socket)
-            bindTargetToLogger(client, server, logger);
+            bindTargetToLogger(client, server, logger)
+        })
+
+        client.socket.once('timeout', () => {
```

Comment:
the event listeners should be added before we do any piping etc. So please move this to before the 2 configureSocketKeepAlive invocations